hack/verify-reports.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
```

Comment:
Is this intended to be run automatically as a presubmit?

Reply:
It's intended to be run as part of the `make verify` target. Some [magic hack](https://github.com/kubernetes-sigs/gateway-api/tree/main/hack/verify-all.sh#L57-L58) in `verify-all.sh` runs all the scripts matching the pattern `verify-*.sh`